pkg/database/realm.go

```diff
@@ -80,20 +80,26 @@ type Realm struct {
 	// Name is the name of the realm.
 	Name string `gorm:"type:varchar(200);unique_index"`
 
+	// RegionCode is both a display attribute and required field for ENX. To
+	// handle NULL and uniqueness, the field is converted from it's ptr type to a
+	// concrete type in callbacks. Do not modify RegionCodePtr directly.
+	RegionCode    string  `gorm:"-"`
+	RegionCodePtr *string `gorm:"column:region_code; type:varchar(10);"`
```

Comment:
does gorm work w/ sql.NullString?

Reply:
Yea, but our code doesn't (really). Changing the type of `RegionCode` requires every caller to update their type. So all cases of:

```go
realm.RegionCode = "foo"
printf(realm.RegionCode)
```

become

```go
if v != "" {
  realm.RegionCode = sql.NullString{Value:v, Valid: true}
} else {
  ream.RegionCode = sql.NullString{}
}

if realm.RegionCode.Valid() {
  printf(realm.RegionCode.Value)
} else {
  printf("")
}
```

and I don't want to put that burden on callers tbh.